identity_diff/src/traits.rs

```diff
@@ -0,0 +1,20 @@
+use serde::{Deserialize, Serialize};
+use std::fmt::Debug;
+
+/// The primary `Diff` Trait type.
+pub trait Diff: Clone + Debug + PartialEq {
+    /// The Corresponding Diff Type for the implemented Type.
+    type Type: Sized + Clone + Debug + PartialEq + for<'de> Deserialize<'de> + Serialize;
+
+    /// Finds the difference between two types; `self` and `other` and returns `Self::Type`
+    fn diff(&self, other: &Self) -> Self::Type;
```

Comment:
These should probably all return `Result<T>` so `DiffError`s can be handled by the user (rather than panic).

something simple

`enum Error { DiffError, MergeError, ConvertError }`